### PR TITLE
Edit and delete comments

### PR DESCRIPTION
**Pull request by RipNasty**
this is just to put it on Git so Therese can pull onto her computer

### CODE REVIEW

**Comment by therese28** [comment]
https://www.loom.com/share/1ae5d3516c9c43a88a872d303c8bcaee

**Comment by AngeC22** [comment]
looks good

**Comment by RipNasty** [comment]
Worked with Therese on this ALL DAY!  Finally got everything to work.

**Comment by Chepenik** [comment]
Amazing work. What a privilege to watch the team struggle with this for three days and come out with some fire code!!!! Feeling energized (no chatGPT did not write this despite my teams questioning)

File: client/src/components/EditForm.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            }
+        );
+        if (response.ok) {
+            setRedirect(true);
+        } else {
+            console.error("Failed to update comment:", response.statusText);
+        }
+        } catch (error) {
+        console.error(`Error in fetch: ${error.message}`);
+        }
```
Comment (nwalberts) [inline client/src/components/EditForm.js:61]: fix indentation

File: client/src/components/EditForm.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        <span {...props}>
+            {state.valueNow}
+            🌶️
+        </span>
+        ),
+    };
+
+    if (redirect) {
+        return <Redirect to={`/websites/${websiteId}`} />
+        }
```
Comment (nwalberts) [inline client/src/components/EditForm.js:80]: fix indentation

File: client/src/components/EditForm.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+   
+    return (
+        <div>
```
Comment (nwalberts) [inline client/src/components/EditForm.js:83]: divs without attributes arent allowed. replace iwth ract fragment

File: client/src/components/EditForm.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            type="text"
+            name="comment"
+            value={commentState.comment}
+            onChange={(event) =>
+                setCommentState({
+                ...commentState,
+                comment: event.target.value,
+                })
+            }
+            />
```
Comment (nwalberts) [inline client/src/components/EditForm.js:105]: fIx iNDeNtaTiOn

File: client/src/components/EditForm.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+            <div className="form-errors">{errors.comment}</div>
+            <button type="submit">Submit</button>
+        </form>
+        </div>
```
Comment (nwalberts) [inline client/src/components/EditForm.js:109]: indentation

File: client/src/components/SocialMediaShow.js [hunk 6]
```diff
@@ -61,6 +61,57 @@ const SocialMediaShow = (props) => {
     }
   }
 
+  const handleCommentDelete = async (commentId) => {
+    try { 
+      const response = await fetch(`/api/v1/websites/${socialMediaShow.id}/comments/${commentId}`, { method: "DELETE" })
```
Comment (nwalberts) [inline client/src/components/SocialMediaShow.js:66]: we should be able to shorten this to `/api/v1//comments/${commentId}` on a regular comments router.

That said, ill allow it, but this would be better with the shorter path, as we only need the id of the comment to delete it

File: package.json [hunk 7]
```diff
@@ -20,6 +20,6 @@
     "node": "^16.18"
   },
   "dependencies": {
-    "react-slider": "^2.0.4"
+
```
Comment (nwalberts) [inline package.json:23]: remove dependencies entirely from this json file

File: server/src/routes/api/v1/commentsRouter.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+})
+
+commentsRouter.patch("/:id", async (req, res) => {
+    const { id } = req.params;
+    try {
+        const commentData = req.body.comment;
+        const comment = await Comment.query().findById(id);
+        const updatedComment = await Comment.query().patch(commentData).findById(id);
+    return res.status(200).json({ comment: updatedComment });
+    } catch (error) {
```
Comment (nwalberts) [inline server/src/routes/api/v1/commentsRouter.js:26]: fix indentation

**Review by nwalberts — CHANGES_REQUESTED** [review]
